content/en/docs/v3.6/dev-internal/discovery_protocol.md

```diff
@@ -33,85 +36,72 @@ UUID=$(uuidgen)
-
-In etcd implementation, the member may check the cluster status even before registering itself. So it could fail quickly if the cluster has been full.
-
-### Waiting for all members
-
-The wait process is described in detail in the [etcd API documentation][api].
-
+The default time or timeout values can also be changed using the following flags, which also follow exactly the same usage as `etcdctl` communicating with an etcd cluster.
 ```
-curl -X GET http://example.com/v2/keys/_etcd/registry/${UUID}?wait=true&waitIndex=${current_etcd_index}
+--discovery-dial-timeout
+--discovery-request-timeout
+--discovery-keepalive-time
+--discovery-keepalive-timeout
 ```
 
-It keeps waiting until finding all members.
-
-## Public discovery service
+### Registering itself
 
-CoreOS Inc. hosts a public discovery service at https://discovery.etcd.io/ , which provides some nice features for ease of use.
+The first thing what each etcd process does is to register itself into the given new cluster as a member. This is done by creating member ID as a key in the full registry key.
 
-### Mask key prefix
+```
+etcdctl --endpoints=http://example.com:2379 put /_etcd/registry/${UUID}/members/${member_id} ${member_name}=${member_peer_url_1}&${member_name}=${member_peer_url_2}
+```
 
-Public discovery service will redirect `https://discovery.etcd.io/${UUID}` to etcd cluster behind for the key at `/v2/keys/_etcd/registry`. It masks register key prefix for short and readable discovery url.
+### Checking the status
 
-### Get new token
+It checks the expected cluster size and registration status, and decides what the next action is.
 
 ```
-GET /new
-
-Sent query:
-	size=${cluster_size}
-Possible status codes:
-	200 OK
-	400 Bad Request
-200 Body:
-	generated discovery url
+etcdctl --endpoints=http://example.com:2379 get /_etcd/registry/${UUID}/_config/size
+etcdctl --endpoints=http://example.com:2379 get /_etcd/registry/${UUID}/members
 ```
 
-The generation process in the service follows the steps from [Creating a New Discovery Token][new-discovery-token] to [Specifying the Expected Cluster Size][expected-cluster-size].
+If registered members are still not enough, it will wait for left members to appear.
 
-### Check discovery status
+If the number of registered members is bigger than the expected size N, it treats the first N registered members as the member list for the cluster. If the member itself is in the member list, the discovery procedure succeeds, and it fetches all peers through the member list. If it is not in the member list, the discovery procedure finishes with the failure that the cluster has been full.
 
-```
-GET /${UUID}
-```
+In etcd implementation, the member may check the cluster status even before registering itself. So it could fail quickly if the cluster has been full.
```

Comment:
We may just want to remove "In etcd implementation," and start the sentence as, "The member may check the..."

Reply:
Thanks & updated.